commands/options_prompt_test.go

```diff
@@ -0,0 +1,167 @@
+		return timestamp
+	}
+
+	tests := []struct {
+		name             string
+		stub             prompterStub
+		expectErr        string
+		expectParameters SupportBundleParameters
+	}{
+		{
+			name: "Include all",
+			stub: prompterStub{
+				includeLogs:          true,
+				includeSystem:        true,
+				includeConfiguration: true,
+				includeThreadDump:    true,
+			},
+			expectParameters: SupportBundleParameters{
+				Configuration: true,
+				Logs: &SupportBundleParametersLogs{
+					Include:   true,
+					StartDate: "2012-10-31",
+					EndDate:   "2012-11-01",
+				},
+				System: true,
+				ThreadDump: &SupportBundleParametersThreadDump{
+					Count:    1,
+					Interval: 0,
+				},
+			},
+		},
+		{
+			name: "Include logs only",
+			stub: prompterStub{
+				includeLogs: true,
+			},
+			expectParameters: SupportBundleParameters{
+				Configuration: false,
+				Logs: &SupportBundleParametersLogs{
+					Include:   true,
+					StartDate: "2012-10-31",
+					EndDate:   "2012-11-01",
+				},
+				System: false,
+				ThreadDump: &SupportBundleParametersThreadDump{
+					Count:    0,
+					Interval: 0,
+				},
+			},
+		},
+		{
+			name: "Include system only",
+			stub: prompterStub{
+				includeSystem: true,
+			},
+			expectParameters: SupportBundleParameters{
+				Configuration: false,
+				Logs: &SupportBundleParametersLogs{
+					Include:   false,
+					StartDate: "2012-10-31",
+					EndDate:   "2012-11-01",
+				},
+				System: true,
+				ThreadDump: &SupportBundleParametersThreadDump{
+					Count:    0,
+					Interval: 0,
+				},
+			},
+		},
+		{
+			name: "Include configuration only",
+			stub: prompterStub{
+				includeConfiguration: true,
+			},
+			expectParameters: SupportBundleParameters{
+				Configuration: true,
+				Logs: &SupportBundleParametersLogs{
+					Include:   false,
+					StartDate: "2012-10-31",
+					EndDate:   "2012-11-01",
+				},
+				System: false,
+				ThreadDump: &SupportBundleParametersThreadDump{
+					Count:    0,
+					Interval: 0,
+				},
+			},
+		},
+		{
+			name: "Include nothing",
+			stub: prompterStub{},
+			expectParameters: SupportBundleParameters{
+				Configuration: false,
+				Logs: &SupportBundleParametersLogs{
+					Include:   false,
+					StartDate: "2012-10-31",
+					EndDate:   "2012-11-01",
+				},
+				System: false,
+				ThreadDump: &SupportBundleParametersThreadDump{
+					Count:    0,
+					Interval: 0,
+				},
+			},
+		},
+		{
+			name: "Error",
+			stub: prompterStub{
+				err: errors.New("oops"),
+			},
+			expectErr: "oops",
+		},
+	}
+	for i := range tests {
+		test := tests[i]
+		t.Run(test.name, func(t *testing.T) {
+			provider := promptOptionsProvider{getDate: clock, prompter: &test.stub}
+			options, err := provider.GetOptions("foo")
+			if test.expectErr != "" {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), test.expectErr)
+			} else {
+				require.NoError(t, err)
+				assert.Equal(t, "JFrog Support Case number foo", options.Name)
```

Comment:
could you use `cmp.Equal` or `cmp.Comparer` to compare `options` with expected?

Reply:
Since name and desc are the same for all tests, I wanted to keep it out of the `cmp.Diff` (so that each test case does not have to repeat the same values)

Reply:
I meant that you could define a `cmp.Comparer` that could be used for all tests. It is basically the same as what is already here, it just wraps the various asserts. This is fine how it is.

Reply:
I'll keep `cmp.Diff` as it has a nice builtin reporting facility.